GVRf/Framework/jni/engine/picker/picker.cpp

```diff
@@ -100,14 +100,12 @@ glm::vec3 Picker::pickSceneObjectAgainstBoundingBox(
         const SceneObject* scene_object, float ox, float oy, float oz, float dx,
         float dy, float dz) {
     glm::mat4 model_matrix = scene_object->transform()->getModelMatrix();
-    Mesh* mesh = scene_object->render_data()->mesh()->getBoundingBox();
-    if (mesh != 0) {
-        EyePointData data = MeshEyePointee::isPointed(mesh, model_matrix, ox,
-                oy, oz, dx, dy, dz);
-        if (data.pointed()) {
-            return data.hit();
-        }
+    std::unique_ptr<Mesh> mesh(scene_object->render_data()->mesh()->createBoundingBox());
```

Comment:
- here the mesh is passed to isPointed which doesn't need to keep it and then discarded; shared_ptr would still work but it would imply possible sharing which is not the case here.
- shared_ptr has to be used everywhere applicable or it won't work